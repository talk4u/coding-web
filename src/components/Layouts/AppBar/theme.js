export const darkTheme = {
    backgroundColor: '#222',
    color: '#e1e1e7'
}

export const defaultTheme = {
    backgroundColor: 'transparent',
    color: '#222'
}