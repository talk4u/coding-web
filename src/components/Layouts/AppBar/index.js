import React from 'react'
import ReactDOM from 'react-dom'

import styled from 'styled-components'
import {appbar_height, max_width} from '../var'
import {darkTheme, defaultTheme} from "./theme";


export const AppBarContainer = styled.div`
    position: fixed;
    width: 100%;
    height: ${appbar_height}px;
    background: ${props=>props.theme.backgroundColor};
    color: ${props=>props.theme.color};
`
const AppBarView = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    max-width: ${max_width}px;
    width: 100%;
    height: 100%;
    margin: 0 auto;
`

const AppBar = ({children}) => {
    return(
        <AppBarContainer>
            <AppBarView>{children}</AppBarView>
        </AppBarContainer>
    )
}

AppBarContainer.defaultProps = {
    theme: defaultTheme
}


export default AppBar