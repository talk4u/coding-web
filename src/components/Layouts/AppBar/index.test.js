import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import AppBar from './index'
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('AppBar Shallow render', () => {
    let container
    beforeEach(()=>{
        container = shallow(<AppBar/> )
    })
    it('+++ render the connected(SMART) component', () => {
        expect(container.length).toEqual(1)
    });
});