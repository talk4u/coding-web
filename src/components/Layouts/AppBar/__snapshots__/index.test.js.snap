// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Shallow render +++ render Styled component 1`] = `
<div
  className="sc-bdVaJa ipOFWd"
/>
`;
